packages/main/src/Tree.js

```diff
@@ -0,0 +1,378 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import TreeItem from "./TreeItem.js";
+import BusyIndicator from "./BusyIndicator.js";
+import List from "./List.js";
+import TreeListItem from "./TreeListItem.js";
+import ListMode from "./types/ListMode.js";
+
+// Template
+import TreeTemplate from "./generated/templates/TreeTemplate.lit.js";
+
+// Styles
+import TreeCss from "./generated/themes/Tree.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-tree",
+	properties: /** @lends sap.ui.webcomponents.main.Tree.prototype */ {
+		/**
+		 * Defines the mode of the <code>ui5-tree</code>. Since the tree uses a <code>ui5-list</code> to display its structure,
+		 * the tree modes are exactly the same as the list modes, and are all applicable.
+		 * <br><br>
+		 * <b>Note:</b> Available options are <code>None</code>, <code>SingleSelect</code>, <code>SingleSelectBegin</code>,
+		 * <code>SingleSelectEnd</code>, <code>MultiSelect</code>, and <code>Delete</code>.
+		 *
+		 * @public
+		 * @type {ListMode}
+		 * @defaultValue "None"
+		 */
+		mode: {
+			type: ListMode,
+			defaultValue: ListMode.None,
+		},
+
+		/**
+		 * Defines the text that is displayed when the <code>ui5-tree</code> contains no items.
+		 *
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 */
+		noDataText: {
+			type: String,
+		},
+
+		/**
+		 * Defines the <code>ui5-tree</code> header text.
+		 * <br><br>
+		 * <b>Note:</b> If <code>header</code> is set this property is ignored.
+		 *
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 */
+		headerText: {
+			type: String,
+		},
+
+		/**
+		 * Defines the <code>ui5-tree</code> footer text.
+		 *
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 */
+		footerText: {
+			type: String,
+		},
+
+		/**
+		 * When set to <code>true</code>, a busy indicator will be displayed on the <code>ui5-tree</code>, blocking any user interaction.
+		 * Use this when you need to dynamically load tree items.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		busy: {
```

Comment:
```ui5-busyindicator``` has default slot which is used for this use case. Do we benefit something from this property?